src/NewSessionView.elm

```diff
@@ -187,65 +185,60 @@ view context model =
-                                [ class "form-control"
+                                [ class "form__input form__input--time-hour-prog-builder"
                                 , type_ "number"
-                                , style [ ( "width", "6rem" ) ]
                                 , value (toStringIgnore0 context.session.startTime.hour)
                                 , onInput UpdateNewSessionStartHour
                                 , placeholder "00"
                                 ]
                                 []
                             , input
-                                [ class "form-control"
+                                [ class "form__input form__input--time-min-prog-builder"
                                 , type_ "number"
-                                , style [ ( "width", "6rem" ) ]
                                 , value (toStringIgnore0 context.session.startTime.minute)
                                 , onInput UpdateNewSessionStartMinute
                                 , placeholder "00"
                                 ]
                                 []
                             ]
                         ]
-                    , div [ class "input-group" ]
-                        [ label []
-                            [ text "End time" ]
+                    , div []
+                        [ label [ class "form__label" ]
+                            [ text "End time *" ]
                         , div []
                             [ input
-                                [ class "form-control"
+                                [ class "form__input form__input--time-hour-prog-builder"
                                 , type_ "number"
-                                , style [ ( "width", "6rem" ) ]
                                 , value (toStringIgnore0 context.session.endTime.hour)
                                 , onInput UpdateNewSessionEndHour
                                 , placeholder "00"
                                 ]
                                 []
                             , input
-                                [ class "form-control"
+                                [ class "form__input form__input--time-min-prog-builder"
                                 , type_ "number"
-                                , style [ ( "width", "6rem" ) ]
                                 , value (toStringIgnore0 context.session.endTime.minute)
                                 , onInput UpdateNewSessionEndMinute
                                 , placeholder "00"
                                 ]
                                 []
                             ]
-                        , div [ style [ ( "margin-top", "1rem" ) ] ] [ text (newSessionViewWarning context model) ]
-                        , div [ style [ ( "margin-top", "1rem" ) ] ]
+                        , div [ class "prog-form--warning" ] [ text (newSessionViewWarning context model) ]
```

Comment:
Furthermore, if it's possible to have this message **only appear** when the user 
1. misses a required field 
2. Clicks save

At the moment it appears before I've had a chance to type anything which can come across as a bit petulant. 
Here is a petulant crab 🦀  . Humans are very quick to anthropomorphise their computers... or think they're broken.